### PR TITLE
[Fix](recycler) recycler need index_suffix and inverted index storage format to delete idx file

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Recycler did not correctly delete idx file for inverted index format V2 and variant data type.

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

File: cloud/src/recycler/recycler.cpp [hunk 1]
```diff
@@ -1273,17 +1274,26 @@ int InstanceRecycler::delete_rowset_data(const doris::RowsetMetaCloudPB& rs_meta
     const auto& rowset_id = rs_meta_pb.rowset_id_v2();
     int64_t tablet_id = rs_meta_pb.tablet_id();
     // process inverted indexes
-    std::vector<int64_t> index_ids;
+    std::vector<std::pair<int64_t, std::string>> index_ids;
     index_ids.reserve(rs_meta_pb.tablet_schema().index_size());
     for (auto& i : rs_meta_pb.tablet_schema().index()) {
-        index_ids.push_back(i.index_id());
+        index_ids.push_back(std::make_pair(i.index_id(), i.index_suffix_name()));
```
Comment (eldenmoon) [inline cloud/src/recycler/recycler.cpp:1280]: will this handle variant subcolumns's indexes?

Comment (airborne12) [reply cloud/src/recycler/recycler.cpp:1280]: index_suffix_name is for variant

File: cloud/src/recycler/recycler.cpp [hunk 2]
```diff
@@ -1273,17 +1274,26 @@ int InstanceRecycler::delete_rowset_data(const doris::RowsetMetaCloudPB& rs_meta
...
+        if (tablet_schema.has_inverted_index_storage_format()) {
+            if (tablet_schema.inverted_index_storage_format() == InvertedIndexStorageFormatPB::V1) {
+                for (const auto& index_id : index_ids) {
+                    file_paths.push_back(inverted_index_path_v1(tablet_id, rowset_id, i,
+                                                                index_id.first, index_id.second));
+                }
+            } else {
+                if (!index_ids.empty()) {
+                    file_paths.push_back(inverted_index_path_v2(tablet_id, rowset_id, i));
+                }
```
Comment (w41ter) [inline cloud/src/recycler/recycler.cpp:1295]:
```suggestion
            } else if (!index_ids.empty()) {
                file_paths.push_back(inverted_index_path_v2(tablet_id, rowset_id, i));
```

**Comment by airborne12** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Review by platoneko — APPROVED** [review]

**Pull request merged**